P1 radiation model for the Linear FV system

freiler opened the pull request:
Refs #31839

freiler committed:
Adding LinearFV Source and Sink term kernels, and the Marshak BC.

freiler committed:
Adding CSV tests for the 1D P1 radiation model in LinearFV

freiler committed:
Adding documentation

grmnptr commented on modules/navier_stokes/doc/content/source/linearfvbcs/LinearFVP1RadiationMarshakBC.md:
```diff
@@ -0,0 +1,27 @@
+# LinearFVP1RadiationMarshakBC
```
@GiudGiud what do you think, should this go here or the heat transfer module?

freiler replied:
Added to the heat transfer module.

grmnptr commented on modules/navier_stokes/doc/content/source/linearfvbcs/LinearFVP1RadiationMarshakBC.md:
```diff
@@ -0,0 +1,27 @@
+
+where:
+
+- $F_G$ is the normal space derivative of the radiation heat flux at the boundaries defined by the `boundary` parameter
```
There is no FG in the equation above.

freiler replied:
Removed

grmnptr commented on modules/navier_stokes/doc/content/source/linearfvkernels/LinearFVP1RadiationSourceSink.md:
```diff
@@ -0,0 +1,24 @@
+The term added reads as follows:
+
+\begin{equation}
+- \int_{\Omega_C} \sigma_a (4\sigma T_{rad}^4 - G) dV
```
What is Omega_C?

grmnptr commented on modules/navier_stokes/doc/content/source/linearfvkernels/LinearFVP1RadiationSourceSink.md:
```diff
@@ -0,0 +1,24 @@
+-  $\sigma_a$ is the absorption coefficient (SI units (1/m))
+-  $\sigma$ is the Stefan-Boltzmann constant (SI units (W/m$^2$/K$^4$))
+-  $T_{rad}$ is the rqadiation temperature (SI units (K))
+
```
Could you add an example?

grmnptr reviewed (approved):
Nice contribution!
We have to decide if it should go to the heat transfer module or if it should stay here.

grmnptr reviewed (changes requested):
Oops

GiudGiud commented on modules/navier_stokes/doc/content/source/linearfvbcs/LinearFVP1RadiationMarshakBC.md:
```diff
@@ -0,0 +1,27 @@
+
+The Marshak boundary condition implemented reads as follows [!citep](modest2021):
+\begin{equation}
+(-\Gamma \nabla G) \cdot \vec{n_b}   = (G - 4 \sigma T_{b, rad}^4) \frac{\epsilon_b}{2 \Gamma (2 - \epsilon_b)}
```
did you figure out where the differences with non-linear FV P1 came from?

freiler replied:
yes, there might be a 4 missing in the HeatConduction::Constants::sigma * Utility::pow<4>(_temperature_radiation) term:

  return _opacity(makeElemArg(_current_elem), determineState()) *
         (_var(makeElemArg(_current_elem), determineState()) -
          HeatConduction::Constants::sigma * Utility::pow<4>(_temperature_radiation(
                                                 makeElemArg(_current_elem), determineState())));

GiudGiud commented on modules/navier_stokes/doc/content/source/linearfvbcs/LinearFVP1RadiationMarshakBC.md:
```diff
@@ -0,0 +1,27 @@
+- $F_G$ is the normal space derivative of the radiation heat flux at the boundaries defined by the `boundary` parameter
+- $G$ is the radiative heat flux (SI units (W/m$^2$))
+- $\sigma$ is the Stefan-Boltzmann constant (SI units (W/m$^2$/K$^4$))
+-  $\sigma_a$ is the absorption coefficient (SI units (1/m))
```
1/m so it might be a cross section in a way?

freiler replied:
it is like an attenuation coefficient

GiudGiud commented on modules/navier_stokes/test/tests/finite_volume/ins/radiation/tests:
```diff
@@ -0,0 +1,28 @@
+    csvdiff = 'rad_isothermal_medium_1d_adiabatic.csv'
+    cli_args = 'wall_temperature=100.0 Outputs/file_base=rad_isothermal_medium_1d_adiabatic'
+    max_threads = 1
+    requirement = 'Solve the 1D isothermal medium with radiation using the P1 formulation and match the analytical solution for adiabatic conditions.'
```
same others

freiler committed:
Changing P1 radiation model to the heat transfer module

freiler committed:
Addressing Peter's reviews.

freiler committed:
Modifying .h headers descriptions

freiler committed:
Adding Robin BC Base class and modifying FunctorRobinBC and MarshakBC accordingly.#31840

Clang format changes

Update modules/heat_transfer/src/linearfvbcs/LinearFVP1RadiationMarshakBC.C

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3376234/), step Clang format on 814c135 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/31840/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/31840/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 7edd10c4e1e441b808e9c6723cac9f12a982381a`

grmnptr commented on modules/heat_transfer/src/linearfvbcs/LinearFVP1RadiationMarshakBC.C:
```diff
@@ -0,0 +1,60 @@
+Real
+LinearFVP1RadiationMarshakBC::getAlpha(Moose::FaceArg face, Moose::StateArg state) const
+{
+
```
```suggestion
```

grmnptr commented on modules/heat_transfer/src/linearfvbcs/LinearFVP1RadiationMarshakBC.C:
```diff
@@ -0,0 +1,60 @@
+Real
+LinearFVP1RadiationMarshakBC::getGamma(Moose::FaceArg face, Moose::StateArg state) const
+{
+
```
```suggestion
```

grmnptr reviewed (changes requested):
Beautiful rework! All good after this and when Guillaume's last comment is addressed.

freiler committed:
Apply suggestions from code review

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3376449/), step Docs: sync website on d2358b4 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/31840/site)

This comment will be updated on new commits.

freiler committed:
Deleting extra tests in navier_stokes module

grmnptr reviewed (approved):
Looks good to me, thanks for doing the abstraction! 
The same tests are hanging on other PRs as well.

GiudGiud commented on modules/heat_transfer/test/tests/radiation_participating_media/rad_isothermal_medium_1d.i:
```diff
@@ -0,0 +1,112 @@
+[]
+
+[Functions]
+  [analytical_sol]
```
so do you ever compare the solution to the analytical sol ?
A PP to compare it could go in the CSV file ensuring we dont increase the error

freiler replied:
I just added a PP to compare the mean value to the analytical solution. Thanks!

GiudGiud reviewed (commented):
do you have comparison against nonlinearFV by any chance?
Measuring the error in the analytical case could suffice tbh

freiler committed:
Adding analytical solution PP

GiudGiud reviewed (approved):
if this does not agree with nonlienarFV we need an issue created asap
